docs/language-server/protocol-language-server.md

```diff
@@ -5179,6 +5182,75 @@ interface ProfilingStopResult {}
 
 None
 
+## AI Operations
+
+### `ai/completion`
+
+TODO: technilcally we already have an `ai/completion` request. Either rename the
+old completion request or make this one v2.
+
+Sent from the client to the server to ask the AI model the code suggestion.
+
+- **Type:** Request
+- **Direction:** Client -> Server
+- **Connection:** Protocol
+- **Visibility:** Public
+
+#### Parameters
+
+```typescript
+interface AiCompletionParameters {
+  prompt: string;
+}
+```
+
+#### Result
+
+```typescript
+type AiCompletionResult = AiCompletionResultSuccess | AiCompletionResultFailure;
+
+interface AiCompletionResultSuccess {
```

Comment:
My take on this is at https://github.com/enso-org/enso/issues/9656#issuecomment-2056646433

> Whatever IDE needs shall be achieved by IDE invoking an Enso library method, just like visualizations and widgets do.

It'd be better if we were moving closer to such _"IDE invokes an Enso method"_ goal and not further.